sdk/cosmos/azure-cosmos/test/query_tests.py

```diff
@@ -214,6 +218,177 @@ def _MockExecuteFunction(self, function, *args, **kwargs):
+        self._validate_query_plan("Select top 10 value count(c.id) from c", created_collection.container_link, 10, [], ['Count'], True, None, None, "None")
+        self._validate_query_plan("Select * from c order by c._ts offset 5 limit 10", created_collection.container_link, None, ['Ascending'], [], False, 5, 10, "None")
+        self._validate_query_plan("Select distinct value c.id from c order by c.id", created_collection.container_link, None, ['Ascending'], [], True, None, None, "Ordered")
+
+    def _validate_query_plan(self, query, container_link, top, order_by, aggregate, select_value, offset, limit, distinct):
+        query_plan_dict = self.client.client_connection._GetQueryPlanThroughGateway(query, container_link)
+        query_execution_info = _PartitionedQueryExecutionInfo(query_plan_dict)
+        self.assertTrue(query_execution_info.has_rewritten_query())
+        self.assertEquals(query_execution_info.has_distinct_type(), distinct != "None")
+        self.assertEquals(query_execution_info.get_distinct_type(), distinct)
+        self.assertEquals(query_execution_info.has_top(), top is not None)
+        self.assertEquals(query_execution_info.get_top(), top)
+        self.assertEquals(query_execution_info.has_order_by(), len(order_by) > 0)
+        self.assertListEqual(query_execution_info.get_order_by(), order_by)
+        self.assertEquals(query_execution_info.has_aggregates(), len(aggregate) > 0)
+        self.assertListEqual(query_execution_info.get_aggregates(), aggregate)
+        self.assertEquals(query_execution_info.has_select_value(), select_value)
+        self.assertEquals(query_execution_info.has_offset(), offset is not None)
+        self.assertEquals(query_execution_info.get_offset(), offset)
+        self.assertEquals(query_execution_info.has_limit(), limit is not None)
+        self.assertEquals(query_execution_info.get_limit(), limit)
+
+    def test_unsupported_queries(self):
+        created_collection = self.config.create_multi_partition_collection_with_custom_pk_if_not_exist(self.client)
+        queries = ['SELECT COUNT(1) FROM c', 'SELECT COUNT(1) + 5 FROM c', 'SELECT COUNT(1) + SUM(c) FROM c']
+        for query in queries:
+            query_iterable = created_collection.query_items(query=query, enable_cross_partition_query=True)
+            try:
+                list(query_iterable)
+                self.fail()
+            except errors.HTTPFailure as e:
+                self.assertEqual(e.status_code, 400)
+
+    def test_query_with_non_overlapping_pk_ranges(self):
+        created_collection = self.config.create_multi_partition_collection_with_custom_pk_if_not_exist(self.client)
+        query_iterable = created_collection.query_items("select * from c where c.pk='1' or c.pk='2'")
+        self.assertListEqual(list(query_iterable), [])
+
+    def test_offset_limit(self):
+        created_collection = self.config.create_multi_partition_collection_with_custom_pk_if_not_exist(self.client)
```

Comment:
Please beef this test up to iterate through different max item counts, so that we know this code works across continuations.

Reply:
cool, makes sense